pkg/controller/connectionsecret/ensuresecret.go

```diff
@@ -50,20 +52,32 @@ func Ensure(client client.Client, namespace, projectName, projectID, clusterName
 }
 
 func fillSecret(secret *corev1.Secret, projectID string, clusterName string, data ConnectionData) error {
-	var connURL, srvConnURL string
+	var connURL, srvConnURL, pvtConnURL, pvtSrvConnURL string
 	var err error
 	if connURL, err = AddCredentialsToConnectionURL(data.ConnURL, data.DBUserName, data.Password); err != nil {
 		return err
 	}
 	if srvConnURL, err = AddCredentialsToConnectionURL(data.SrvConnURL, data.DBUserName, data.Password); err != nil {
 		return err
 	}
+	if data.PvtConnURL == "" {
+		pvtConnURL = ""
```

Comment:
In both cases strings are empty by default. We can check the opposite and add credentials only if they are not empty:
```
if data.PvtConnURL != "" {
    if pvtConnURL, err = AddCredentialsToConnectionURL(data.PvtConnURL, data.DBUserName, data.Password); err != nil {
        return err
    }
}
```

Reply:
I like it! Haven't had the pleasure of truly jumping into golang yet so still learning some of those basics. Thank you for the suggestion!